### PR TITLE
Fix funcdef lookup in auto_mixed_precision grappler pass

### PR DESCRIPTION
**Pull request by benbarsdell**
Non-inlined funcdefs only exist in the graph's function registry instead of the global op registry. Such ops (e.g., tf.nn.swish) previously caused the auto_mixed_precision grappler pass to fail.

This commit also adds a test for such cases.

attn @reedwm 
cc @nluehr

### CODE REVIEW

**Review by reedwm — APPROVED** [review]

**Comment by cklsoft** [comment]
@benbarsdell Will auto_mixed_precision do the  automatic loss scaling?

**Comment by nluehr** [comment]
Loss scaling is not implemented within the grappler graph optimizer. Setting `auto_mixed_precision=True` in ConfigProto will convert FP32 ops in your graph to fp16, but will not insert the additional operations to implement loss scaling.

You can enable both loss scaling and the graph optimizer by wrapping your optimizer in `tf.train.experimental.enable_mixed_precision_graph_rewrite()`. See https://www.tensorflow.org/api_docs/python/tf/train/experimental/enable_mixed_precision_graph_rewrite for details.
